pkg/operator/gcwatchercontroller/gcwatcher_controller.go

```diff
@@ -135,6 +156,19 @@ func (c *GarbageCollectorWatcherController) sync(ctx context.Context, syncCtx fa
 	if err != nil { // Could be intermittent issues with connectivity, try after sometime, don't set the status yet.
 		return err
 	}
+	progressingMonitoringCond := clusteroperatorhelpers.FindStatusCondition(monitoringClusterOperator.Status.Conditions, "Progressing")
+	// If we just started cluster monitoring stack rollout
+	// Time-out after one hour in case cluster monitoring operator gets stuck, so we can report degraded KCM
+	if (progressingMonitoringCond != nil &&
+		progressingMonitoringCond.Status == configv1.ConditionTrue &&
+		progressingMonitoringCond.LastTransitionTime.After(time.Now().Add(-monitoringStackDeployTimeout))) ||
+		(progressingMonitoringCond == nil && monitoringClusterOperator.CreationTimestamp.After(time.Now().Add(-monitoringStackDeployTimeout))) {
+		// To prevent degradation of KCM when installing the cluster monitoring stack or when a new version of cluster monitoring is being rolled out
+		klog.V(5).Info("Monitoring is being rolled out in the cluster and a diagnostic of the garbage collector is not available at this moment. Please look at the kcm logs for more information to debug the garbage collector further")
```

Comment:
nit: "the kcm logs" may be enough for folks who spend time in kube-controller-manager space.  And this line will show up in the operator,  so maybe even folks who aren't aware of the acronym can figure it out.  But it would be nice if this reference could be more explicit.  Something like:

> Please see 'oc -n openshift-kube-controller-manager logs -l app=kube-controller-manager --tail -1' for more information to debug the garbage collector.

Reply:
I have updated the message to expand the shortcut to `kube-controller-manager`. I would not prefer to add more detailed debugging information since it is hard coding the deployment of KCM. Also, somebody who is inspecting the KCM operator logs, probably knows  how to inspect KCM logs as well.